### PR TITLE
Fixes #3337 Unable to retrieve guest permissions while logged in

### LINKED ISSUE #3337: Unable to retrieve guest permissions while logged in
As the current `user_permissions()` function resets the uid to current user's uid if the uid is 0; it is impossible to get the user permission of a guest while logged in.

Problem instance:
https://community.mybb.com/thread-218439-post-1308791.html#pid1308791

### PR DESCRIPTION
**Pull request by effone**
Attempt to fix #3337 
Guest group permission fix

### CODE REVIEW

File: inc/functions.php [hunk 1]
```diff
@@ -1206,16 +1206,22 @@ function fetch_page_url($url, $page)
  * @param int $uid The user ID
```
Comment (euantorano) [inline inc/functions.php]: Could you update this comment to mention that passing `null` will use the current user's ID?

Comment (effone) [reply inc/functions.php]: Done.

**Comment by euantorano** [comment]
As this isn’t a bug introduced by MyBB 1.8.16 I’m going to postpone merging until 1.8.18 so that we don’t delay 1.8.17 any further.

**Comment by euantorano** [comment]
Appears to work fine for me.

**Pull request merged**